pkgs/applications/networking/eddie-ui/default.nix

```diff
@@ -0,0 +1,92 @@
+{ stdenv,
+  fetchFromGitHub,
+  clang,
+  cmake,
+  mono,
+  dotnet-sdk,
+  which,
+  glibc,
+  pkgconfig,
+  gtk2,
+  libappindicator-gtk2,
+  lzma,
+  libselinux,
+  libsepol,
+  fribidi,
+  libthai,
+  libdatrie,
+  xorg,
+  openvpn,
+  stunnel
+  }:
+
+stdenv.mkDerivation rec {
+  pname = "eddie-ui";
+  version = "2.18.9";
+  arch = if stdenv.system == "i686-linux"
+    then "x86"
+    else "x64";
+  src = fetchFromGitHub {
+    owner = "AirVPN";
+    repo = "Eddie";
+    rev = "2.18.9";
+    sha256 = "1nvnvy6lskydfrlsrbyk065hl6zxdyivhrynydny1kirz3qmrw61";
+  };
+  sourceRoot= "/build/source/src/UI.GTK.Linux.Tray";
+  buildPhase = ''
+    chmod -R +w /build/source
+    cd /build/source/src/
+    find ./ -type f -iname "*.sh" -exec chmod +x {} \;
+    patchShebangs ./
+
+    # Compile C# sources
+    cd /build/source
+    xbuild /verbosity:minimal /p:Configuration="Release" /p:Platform="$arch" src/eddie2.linux.sln
+  
+    # Compile C sources (Tray)
+    export NIX_CFLAGS_COMPILE="$(< ${clang}/nix-support/libcxx-cxxflags) $NIX_CFLAGS_COMPILE" # without this can't find stdlib.h
+    cd src/UI.GTK.Linux.Tray
+    substituteInPlace CMakeLists.txt --replace '/var/empty/include/libappindicator-0.1' '${libappindicator-gtk2.dev}/include/libappindicator-0.1';
+    ./build-$arch.sh
+    cd ../..
+    # Compile C sources
+    sed -i '35 s/-static//' src/App.CLI.Linux.Elevated/build.sh # doesn't build with '-static' option
+    src/eddie.linux.postbuild.sh "src/App.Forms.Linux/bin/$arch/Release/" ui $arch Release
+  '';   
+  nativeBuildInputs = [pkgconfig openvpn mono stunnel];
+  buildInputs = [ which cmake dotnet-sdk gtk2 lzma.dev glibc libappindicator-gtk2.dev  
+                    libselinux libsepol fribidi libthai libdatrie xorg.libXdmcp
+  ]; 
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    mkdir -p $out/lib/eddie-ui
+    cp repository/linux_arch/bundle/eddie-ui/usr/bin/eddie-ui $out/bin/eddie-ui
+    substituteInPlace $out/bin/eddie-ui --replace '/usr/' $out/
+    chmod +x $out/bin/eddie-ui
```

Comment:
```suggestion
    install -Dm755 repository/linux_arch/bundle/eddie-ui/usr/bin/eddie-ui $out/bin/eddie-ui
    substituteInPlace $out/bin/eddie-ui --replace '/usr/' $out/
```

Reply:
Regarding this part. I wanted to ask maybe there is a better way instead of copying Arch script. The script is:
`mono /usr/lib/eddie-ui/eddie-ui.exe --path.resources=/usr/share/eddie-ui --path.exec="$0" "$@"`.
Can we use some Nix wrapper function here?

Reply:
I am not sure but it could be that the binary gets double wrapped.

Reply:
makeWrapper has an ``--add-flags`` option which can do that.